### PR TITLE
Danger is now a peer dependency

### PR DESCRIPTION
**Pull request by stellarhoof**
Also updated the code to account for a breaking change in Danger's API (or github's I'm not sure). This was necessary to keep compatibility with all `^6.0.0` versions.

### CODE REVIEW

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Danger failed to run `/home/circleci/project/src/dangerfile.js`.
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>:exclamation: This PR is BIG (+890 -1485)  
Please keep it below 500 net changes</td>
    </tr>
  </tbody>
</table>


## Error TypeError
```
Cannot read property 'length' of null
TypeError: Cannot read property 'length' of null
    at noPrDescription (/home/circleci/project/src/duties/pr.js:24:29)
    at n (/home/circleci/project/node_modules/lodash/lodash.min.js:5:89)
    at /home/circleci/project/node_modules/lodash/lodash.min.js:51:79
    at c (/home/circleci/project/node_modules/lodash/lodash.min.js:6:348)
    at /home/circleci/project/node_modules/lodash/lodash.min.js:51:56
    at n (/home/circleci/project/node_modules/lodash/lodash.min.js:5:89)
    at /home/circleci/project/node_modules/lodash/lodash.min.js:63:257
    at schedule (/home/circleci/project/src/dangerfile.js:27:46)
    at <anonymous>
```
### Dangerfile
```
19|     let { config } = await explorer.load(process.cwd())
20|     dutiConfig = config
21|   } catch (e) {
22|     dutiConfig = defaultConfig
23|   }
-------------------------------^
24| 
25|   let config = dutiConfig
26| 
27|   return Promise.all(
```
  
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 8cabf83eaffbb768e3487e727573bf46d285dae0
</p>

File: src/duties/pr.js [hunk 1]
```diff
@@ -30,7 +30,7 @@ let bigPr = ({ danger, warn, config: { prNetChangeThreshold } }) => {
 }
 
 let noPrDescription = ({ danger, fail }) => {
-  if (danger.github.pr.body.length === 0) {
+  if (danger.github.pr.body === null) {
```
Comment (daedalus28) [inline src/duties/pr.js]: This seems unrelated, but either way we should probably check both:
https://github.com/artsy/peril-settings/blob/master/org/allPRs.ts#L20

**Comment by stellarhoof** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Danger failed to run `dangerfile.js`.
  </td>
    </tr>
  </tbody>
</table>



## Error undefined
```
undefined
undefined
```
### Dangerfile
```
require('./src/dangerfile')

```
  
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 70d39282af207d467e771663d1d7ded008e64efe
</p>

File: package.json [hunk 2]
```diff
@@ -51,5 +51,8 @@
     "eslint-config-smartprocure": "^1.2.0",
     "jest": "^20.0.4",
     "prettier": "^1.5.3"
+  },
+  "peerDependencies": {
+    "danger": "^6.0.0"
```
[the remark this answers is not in the document]
Comment (stellarhoof) [reply package.json]: It was already a dependency so I left it there. Should I move it to dev dependencies instead?

**Pull request merged**